Begin moving functions from LocalFrameView to FrameView

charliewolfe opened the pull request:
#### 10d1e7485040a2f6ec54279ad9d7fe6dad7b5c44
<pre>
Begin moving functions from LocalFrameView to FrameView
<a href="https://bugs.webkit.org/show_bug.cgi?id=265709">https://bugs.webkit.org/show_bug.cgi?id=265709</a>
<a href="https://rdar.apple.com/119058105">rdar://119058105</a>

Reviewed by Alex Christensen.

* Source/WebCore/Sources.txt:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::ownerRenderer const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/FrameView.cpp: Added.
(WebCore::FrameView::headerHeight const):
(WebCore::FrameView::footerHeight const):
(WebCore::FrameView::topContentInset const):
(WebCore::FrameView::visibleContentScaleFactor const):
(WebCore::FrameView::isActive const):
(WebCore::FrameView::enclosingScrollableArea const):
(WebCore::FrameView::invalidateRect):
(WebCore::FrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebCore::FrameView::scrollableAreaBoundingBox const):
(WebCore::FrameView::hostWindow const):
(WebCore::FrameView::scrollbarStyleChanged):
(WebCore::FrameView::scrollAnimatorEnabled const):
(WebCore::FrameView::convertFromRendererToContainingView const):
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::ownerRenderer const): Deleted.
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::invalidateRect): Deleted.
(WebCore::LocalFrameView::headerHeight const): Deleted.
(WebCore::LocalFrameView::footerHeight const): Deleted.
(WebCore::LocalFrameView::topContentInset const): Deleted.
(WebCore::LocalFrameView::hostWindow const): Deleted.
(WebCore::LocalFrameView::isActive const): Deleted.
(WebCore::LocalFrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const): Deleted.
(WebCore::LocalFrameView::visibleContentScaleFactor const): Deleted.
(WebCore::LocalFrameView::enclosingScrollableArea const): Deleted.
(WebCore::LocalFrameView::scrollableAreaBoundingBox const): Deleted.
(WebCore::LocalFrameView::scrollbarStyleChanged): Deleted.
(WebCore::LocalFrameView::scrollAnimatorEnabled const): Deleted.
(WebCore::LocalFrameView::convertFromRendererToContainingView const): Deleted.
(WebCore::LocalFrameView::convertFromContainingViewToRenderer const): Deleted.
(WebCore::LocalFrameView::convertToContainingView const): Deleted.
(WebCore::LocalFrameView::convertFromContainingView const): Deleted.
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/RemoteFrameView.cpp:
(WebCore::RemoteFrameView::invalidateRect): Deleted.
(WebCore::RemoteFrameView::isActive const): Deleted.
(WebCore::RemoteFrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const): Deleted.
(WebCore::RemoteFrameView::enclosingScrollableArea const): Deleted.
(WebCore::RemoteFrameView::scrollableAreaBoundingBox const): Deleted.
(WebCore::RemoteFrameView::hostWindow const): Deleted.
* Source/WebCore/page/RemoteFrameView.h:

Canonical link: <a href="https://commits.webkit.org/271493@main">https://commits.webkit.org/271493@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/42c2a97c4ab7fed4de327260bddcdfda8033ff2c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28595 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7240 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/29978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [❌ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/29103 "Failed to checkout and rebase branch from PR 21199") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9353 "Built successfully") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4610 "Failed to checkout and rebase branch from PR 21199") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/31242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28865 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/5984 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/29978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/5356 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/29978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/32561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/26184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/29978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/32561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/5321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/51/builds/4610 "Failed to checkout and rebase branch from PR 21199") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/32561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/6927 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/29978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/5782 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3690 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/5840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -0,0 +1,345 @@
+    renderer->repaintRectangle(repaintRect);
+}
+
+bool FrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting() const
```
Probably in a different PR, but this should probably be renamed to scrollingPerformanceTestingEnabled.  It looks like it used to do something, but now it just returns that boolean from the page.

achristensen07 commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -0,0 +1,345 @@
+
+    ScrollView::scrollbarStyleChanged(newStyle, forceUpdate);
+}
+
```
extra space.

charliewolfe committed:
Begin moving functions from LocalFrameView to FrameView
https://bugs.webkit.org/show_bug.cgi?id=265709
rdar://119058105

Reviewed by Alex Christensen.

* Source/WebCore/Sources.txt:
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::ownerRenderer const):
* Source/WebCore/page/Frame.h:
* Source/WebCore/page/FrameView.cpp: Added.
(WebCore::FrameView::headerHeight const):
(WebCore::FrameView::footerHeight const):
(WebCore::FrameView::topContentInset const):
(WebCore::FrameView::visibleContentScaleFactor const):
(WebCore::FrameView::isActive const):
(WebCore::FrameView::enclosingScrollableArea const):
(WebCore::FrameView::invalidateRect):
(WebCore::FrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const):
(WebCore::FrameView::scrollableAreaBoundingBox const):
(WebCore::FrameView::hostWindow const):
(WebCore::FrameView::scrollbarStyleChanged):
(WebCore::FrameView::scrollAnimatorEnabled const):
(WebCore::FrameView::convertFromRendererToContainingView const):
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/page/InteractionRegion.cpp:
(WebCore::interactionRegionForRenderedRegion):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::ownerRenderer const): Deleted.
* Source/WebCore/page/LocalFrame.h:
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::invalidateRect): Deleted.
(WebCore::LocalFrameView::headerHeight const): Deleted.
(WebCore::LocalFrameView::footerHeight const): Deleted.
(WebCore::LocalFrameView::topContentInset const): Deleted.
(WebCore::LocalFrameView::hostWindow const): Deleted.
(WebCore::LocalFrameView::isActive const): Deleted.
(WebCore::LocalFrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const): Deleted.
(WebCore::LocalFrameView::visibleContentScaleFactor const): Deleted.
(WebCore::LocalFrameView::enclosingScrollableArea const): Deleted.
(WebCore::LocalFrameView::scrollableAreaBoundingBox const): Deleted.
(WebCore::LocalFrameView::scrollbarStyleChanged): Deleted.
(WebCore::LocalFrameView::scrollAnimatorEnabled const): Deleted.
(WebCore::LocalFrameView::convertFromRendererToContainingView const): Deleted.
(WebCore::LocalFrameView::convertFromContainingViewToRenderer const): Deleted.
(WebCore::LocalFrameView::convertToContainingView const): Deleted.
(WebCore::LocalFrameView::convertFromContainingView const): Deleted.
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/RemoteFrameView.cpp:
(WebCore::RemoteFrameView::invalidateRect): Deleted.
(WebCore::RemoteFrameView::isActive const): Deleted.
(WebCore::RemoteFrameView::forceUpdateScrollbarsOnMainThreadForPerformanceTesting const): Deleted.
(WebCore::RemoteFrameView::enclosingScrollableArea const): Deleted.
(WebCore::RemoteFrameView::scrollableAreaBoundingBox const): Deleted.
(WebCore::RemoteFrameView::hostWindow const): Deleted.
* Source/WebCore/page/RemoteFrameView.h:

Canonical link: https://commits.webkit.org/271493@main
